docs/src/main/java/org/geotools/wmts/WMTSCoverage.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ */
+
+package org.geotools.wmts;
+
+import java.awt.Rectangle;
+import java.io.File;
+import java.net.URL;
+import javax.imageio.ImageIO;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridEnvelope2D;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.ows.wmts.WebMapTileServer;
+import org.geotools.ows.wmts.map.WMTSCoverageReader;
+import org.geotools.ows.wmts.model.WMTSLayer;
+import org.geotools.parameter.Parameter;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.opengis.parameter.GeneralParameterValue;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * Uses the WMTSCoverageReader to write a mosaicked image of tiles covering a given extent.
+ *
+ * <p>The extent is given in a projection that is not provided by the server.
+ *
+ * <p>You must give a path to where the file should be saved. It will be in PNG.
+ *
+ * @author Roar Brænden
+ */
+public class WMTSCoverage {
+
+    private static final double[] LOMFJORDBOTNEN_EXTENT = {17.0697, 18.1937, 79.3052, 79.4667};
+    private static final String WMTS_ORTOFOTO_URL =
+            "https://geodata.npolar.no/arcgis/rest/services/Basisdata/NP_Ortofoto_Svalbard_WMTS_25833/MapServer/WMTS/1.0.0/WMTSCapabilities.xml";
+
+    public static void main(String[] args) throws Exception {
+
+        if (args.length < 1) {
+            throw new IllegalArgumentException("You must provide a path for the resulting image.");
+        }
+
+        File resultFile = new File(args[0]);
+
+        WebMapTileServer wmts = new WebMapTileServer(new URL(WMTS_ORTOFOTO_URL));
```

Comment:
For code snippets I often use marker comments and then use sphinx to include the code between the comments.